pkg/cmd/admin/node/node.go

```diff
@@ -0,0 +1,399 @@
+	})
+	if len(errList) != 0 {
+		return fmt.Errorf("%v", errList)
+	}
+	if nodeCount == 0 {
+		if len(cfg.NodeNames) > 0 {
+			return fmt.Errorf("Nodes %v not found", strings.Join(cfg.NodeNames, ","))
+		} else {
+			return fmt.Errorf("No nodes found")
+		}
+	}
+	return nil
+}
+
+func (cfg *NodeConfig) RunSchedulable(node *kapi.Node, ignoreHeaders bool) error {
+	if cfg.Schedulable {
+		if node.Spec.Unschedulable {
+			node.Spec.Unschedulable = false
+		}
+	} else {
+		if !node.Spec.Unschedulable {
+			node.Spec.Unschedulable = true
+		}
+	}
+
+	updatedNode, err := cfg.Kclient.Nodes().Update(node)
+	if err != nil {
+		return err
+	}
+
+	printerWithHeaders, printerNoHeaders, err := cfg.getPrinters("", updatedNode)
+	if err != nil {
+		return err
+	}
+	if ignoreHeaders {
+		printerNoHeaders.PrintObj(updatedNode, cfg.Writer)
+	} else {
+		printerWithHeaders.PrintObj(updatedNode, cfg.Writer)
+	}
+	return nil
+}
+
+func (cfg *NodeConfig) RunEvacuate(node *kapi.Node) error {
+	if !node.Spec.Unschedulable {
+		return fmt.Errorf("Node '%s' must be unschedulable to perform evacuation.\nYou can mark the node unschedulable with openshift admin manage-node %s --schedulable=false", node.ObjectMeta.Name, node.ObjectMeta.Name)
+	}
+
+	selector, err := labels.Parse(cfg.PodSelector.Value())
+	if err != nil {
+		return err
+	}
+
+	pods, err := cfg.Kclient.Pods(kapi.NamespaceAll).List(selector, fields.Everything())
+	if err != nil {
+		return err
+	}
+	rcs, err := cfg.Kclient.ReplicationControllers(kapi.NamespaceAll).List(labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	printerWithHeaders, printerNoHeaders, err := cfg.getPrinters("pod", nil)
+	if err != nil {
+		return err
+	}
+
+	errList := []error{}
+	firstPod := true
+	// grace = 0 implies delete the pod immediately
+	grace := int64(0)
+	deleteOptions := &kapi.DeleteOptions{GracePeriodSeconds: &grace}
+
+	for _, pod := range pods.Items {
+		if pod.Spec.Host != node.ObjectMeta.Name {
+			continue
+		}
+
+		foundrc := false
+		for _, rc := range rcs.Items {
+			selector := labels.SelectorFromSet(rc.Spec.Selector)
+			if selector.Matches(labels.Set(pod.Labels)) {
+				foundrc = true
+				break
+			}
+		}
+
+		if firstPod {
+			fmt.Fprintln(cfg.Writer, "\nMigrating these pods on node: ", node.ObjectMeta.Name, "\n")
+			firstPod = false
+			printerWithHeaders.PrintObj(&pod, cfg.Writer)
+		} else {
+			printerNoHeaders.PrintObj(&pod, cfg.Writer)
+		}
+
+		if err := cfg.Kclient.Pods(pod.Namespace).Delete(pod.Name, deleteOptions); err != nil {
+			glog.Errorf("Unable to delete a pod: %+v, error: %v", pod, err)
+			errList = append(errList, err)
+			continue
+		}
+		if !foundrc {
+			// For pods without replication controller, pod.Spec.Host set by user will not be honored.
+			// more details: https://github.com/GoogleCloudPlatform/kubernetes/issues/8007
+			pod.Spec.Host = ""
+			pod.ObjectMeta.ResourceVersion = ""
+			pod.Status = kapi.PodStatus{}
+			_, err := cfg.Kclient.Pods(pod.Namespace).Create(&pod)
```

Comment:
Hmm, since this is an admin command I'd expect it to be run by a cluster admin who should have elevated privileges anyway, right?  If you're managing a node I would think you're operating with the highest level of privileges.